tipg/collections.py

```diff
@@ -961,23 +951,27 @@ async def get_collection_index(  # noqa: C901
             table_conf = table_confs.get(confid, TableConfig())
 
             # Make sure that any properties set in conf exist in table
-            properties = sorted(table.get("properties", []), key=lambda d: d["name"])
-            properties_setting = table_conf.properties or []
-            if properties_setting:
-                properties = [p for p in properties if p["name"] in properties_setting]
+            columns = sorted(table.get("properties", []), key=lambda d: d["name"])
+            properties_setting = table_conf.properties or [c["name"] for c in columns]
 
             # ID Column
-            id_column = table_conf.pk or table.get("pk")
-            if not id_column and fallback_key_names:
-                for p in properties:
+            id_column = None
+            if id_name := table_conf.pk or table.get("pk"):
+                for p in columns:
+                    if id_name == p["name"]:
+                        id_column = p
+                        break
+
+            if id_column is None and fallback_key_names:
+                for p in columns:
                     if p["name"] in fallback_key_names:
-                        id_column = p["name"]
+                        id_column = p
                         break
 
             datetime_column = None
             geometry_column = None
 
-            for c in properties:
+            for c in columns:
```

Comment:
⚠️ we use the `full` columns instead of the filtered properties 

I feel this is a better way to do, if user want to change the geometry or datetime column we have to use the `geomcol` or `datetimecol` table config instead of filtering the properties